bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/MetadataSelectorMatcher.java

```diff
@@ -72,19 +63,22 @@ public MetadataSelectorMatcher(final @Reference MetadataRegistry metadataRegistr
                     .map(n -> n.trim()) //
                     .collect(Collectors.toSet());
 
+            Set<String> allMetadataNamespaces = metadataRegistry.getAll().stream() //
+                    .map(metadata -> metadata.getUID().getNamespace()) //
+                    .distinct() //
+                    .collect(Collectors.toSet());
```

Comment:
Use shortcut here or remove static import.

```suggestion
                    .collect(toSet());
```